src/main/java/com/example/pet_shelter/controller/AnimalRESTController.java

```diff
@@ -16,33 +17,38 @@ public class AnimalRESTController {
     private AnimalService animalService;
 
     @GetMapping("/animals")
+    @Cacheable(value = "ShowingAllAnimals")
     public List<Animal> showAllEmployees() {
```

Comment:
not worry good idea add caching for getAll controller (you animal will be updated or new animal will be added)